File: DownloadFiles/TGNet/Chapter06/VB/Exercise6-7/TestHarness/TestHarness/Form1.vb
Imports System.Text.RegularExpressions
Public Class Form1

  Private Sub btnTest_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnTest.Click

    Dim objReport As New ReportMaker.ReportToFile

    objReport.Tester = "Joe tester"
    objReport.ApplicationName = "Test App"
    objReport.BuildNumber = "1.1"
    objReport.DateReported = Date.Today.ToShortDateString()
    objReport.Description = "This is a test Bug"
    objReport.FileName = txtFileName.Text

    Dim strTempString As String

    'Convert to lowercase to make the RegEx simpler
    strTempString = txtFileName.Text.ToLower()

    Dim strXMLPattern As String = "^.+\.[x][m][l]$"
    Dim strHTMLPattern As String = "^.+\.[h][t][m][l]$"

    Try

      If (Regex.IsMatch(strTempString, strXMLPattern)) Then
        objReport.FormatData("xml")
      ElseIf (Regex.IsMatch(strTempString, strHTMLPattern)) Then
        objReport.FormatData("html")
      Else
        objReport.FormatData()
      End If
      objReport.Save()
      MessageBox.Show("Saved")

    Catch ex As Exception
      MessageBox.Show(ex.ToString)
    End Try

  End Sub
End Class

File: DownloadFiles/TGNet/Chapter06/C#/Exercise6-7/TestHarness/TestHarness/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.Text.RegularExpressions;
namespace TestHarness
{
  public partial class Form1 : Form
  {
    public Form1()
    {
      InitializeComponent();
    }

   private void btnTest_Click(object sender, EventArgs e)
    {
      ReportMaker.ReportToFile  objReport = new ReportMaker.ReportToFile();
      objReport.Tester = "Joe tester";
      objReport.ApplicationName = "Test App";
      objReport.BuildNumber = "1.1";
      objReport.DateReported = System.DateTime.Today.ToShortDateString();
      objReport.Description = "This is a test Bug";
      objReport.FileName = txtFileName.Text;
      string strTempString;
      strTempString = txtFileName.Text.ToLower();
      string strXMLPattern = "^.+\\.[x][m][l]$";
      string strHTMLPattern = "^.+\\.[h][t][m][l]$";
      try
      {
        if ((Regex.IsMatch(strTempString, strXMLPattern)))
        {
          objReport.FormatData("xml");
        }
        else if ((Regex.IsMatch(strTempString, strHTMLPattern)))
        {
          objReport.FormatData("html");
        }
        else
        {
          objReport.FormatData();
        }
        objReport.Save();
        MessageBox.Show("Saved");
      }
      catch (Exception ex)
      {
        MessageBox.Show(ex.ToString());
      }
    }//end of btnTest_Click

  }
}